src/System.Runtime/tests/Performance/Perf.String.cs

```diff
@@ -0,0 +1,243 @@
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        [InlineData(10)]
+        [InlineData(100)]
+        public void Format(int numberOfObjects)
+        {
+            // Setup the format string and the list of objects to format
+            StringBuilder formatter = new StringBuilder();
+            List<string> objects = new List<string>();
+            for (int i = 0; i < numberOfObjects; i++)
+            {
+                formatter.Append("%s, ");
+                objects.Add(PerfUtils.CreateString(10));
+            }
+            string format = formatter.ToString();
+            string[] objectArr = objects.ToArray();
+
+            // Perform the actual formatting
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    string.Format(format, objectArr);
+        }
+
+        [Benchmark]
+        [MemberData("TestStringSizes")]
+        public void GetLength(int size)
+        {
+            int result;
+            string testString = PerfUtils.CreateString(size);
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                {
+                    result = testString.Length; result = testString.Length; result = testString.Length;
+                    result = testString.Length; result = testString.Length; result = testString.Length;
+                    result = testString.Length; result = testString.Length; result = testString.Length;
+                }
+        }
+
+        [Benchmark]
+        [MemberData("TestStringSizes")]
+        public void op_Equality(int size)
+        {
+            bool result;
+            string testString1 = PerfUtils.CreateString(size);
+            string testString2 = PerfUtils.CreateString(size);
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                {
+                    result = testString1 == testString2; result = testString1 == testString2;
+                    result = testString1 == testString2; result = testString1 == testString2;
+                    result = testString1 == testString2; result = testString1 == testString2;
+                }
+        }
+
+        [Benchmark]
+        [MemberData("TestStringSizes")]
+        public void Replace(int size)
+        {
+            string testString = PerfUtils.CreateString(size);
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    testString.Replace("-", "1");
+        }
+
+        [Benchmark]
+        [MemberData("TestStringSizes")]
+        public void Split(int size)
+        {
+            string testString = PerfUtils.CreateString(size);
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    testString.Split("-");
```

Comment:
Presumably if you end up changing the string generation to not use Guid.NewGuid, some of these string tests will need to change as well.